src/auth/auth.service.ts

```diff
@@ -14,98 +21,132 @@ export class AuthService {
     constructor(
         private readonly userService: UserService,
         private readonly jwtService: JwtService
-    ) { }
+    ) {}
 
     /**
      * Register User (Creates a new one)
      * @param credentials of the user
      * @returns the new registered User
      */
-    async registerUser(credentials: RegisterDto): Promise<any> {
-        //While this might seem unnecessary now, this way of implementing this allows us to add logic to register later without affecting the user create itself  
-        const user: User = await this.userService.create(credentials)
+    async registerUser(credentials: RegisterDto): Promise<AccessTokenDto> {
+        // While this might seem unnecessary now, this way of implementing this allows us to add logic to register later without affecting the user create itself
+        const user: User = await this.userService.create(credentials);
 
+        /* istanbul ignore next */
         if (!user)
-            new BadRequestException()
+            new InternalServerErrorException('User could not be created');
 
-        return await this.createLoginPayload(user)
+        // Generate and return JWT
+        return await this.createLoginPayload(user);
     }
 
     /**
      * Search for a user by username and validate with the password
      * @param username of the user
      * @param password of the user
-     * @returns user without password or if user do not exist returns null 
+     * @returns user without password or if user do not exist returns null
      */
-    async validateUserWithEmailPassword(email: string, password: string): Promise<any> {
-        const user = await this.userService.findOneByEmail(email)
+    async validateUserWithEmailPassword(
+        email: string,
+        password: string
+    ): Promise<User> {
+        const user = await this.userService.findOneByEmail(email);
         if (!user)
-            throw new BadRequestException(`There is no user with the email: ${email}`)
-
-        if (user.provider)
-            throw new UnauthorizedException(`User can't login with Username and Password, already has account on ${user.provider}`)
+            throw new UnauthorizedException(
+                `Login Failed due to invalid credentials`
+            );
+
+        if (user.provider) {
+            throw new UnauthorizedException(
+                `Login Failed due to invalid credentials. There might be a third party login with this email`
+            );
+        }
 
         if (await bcrypt.compare(password, user.password)) {
```

Comment:
We do like
1. User is not found => Login failed
2. Provider login => Login failed
3. Password okay => Login success
4. Okay password wrong => Login failed

Maybe do all validation together and return user at the end:
1. User is not found => Login failed
2. Provider login => Login failed
3. Password is not okay => Login failed
// We can insert more here when ever we want 
4. ALL GOOD => Login success

Reply:
done